src/main/java/bio/terra/controller/RepositoryApiController.java

```diff
@@ -120,9 +104,33 @@ protected void initBinder(final WebDataBinder binder) {
     }
 
     public ResponseEntity<StudyModel> retrieveStudy(@PathVariable("id") String id) {
-        Study study = studyDao.retrieve(UUID.fromString(id));
-        StudyModel studyModel = StudyJsonConversion.studyModelFromStudy(study);
-        return new ResponseEntity<>(studyModel, HttpStatus.OK);
+        return new ResponseEntity<>(studyService.retrieve(UUID.fromString(id)), HttpStatus.OK);
+    }
+
+    public ResponseEntity<Void> deleteStudy(@PathVariable("id") String id) {
+        boolean success = studyService.delete(UUID.fromString(id));
+        return success ? new ResponseEntity<>(HttpStatus.OK) : new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    public ResponseEntity<List<StudySummaryModel>> enumerateStudies(Integer offset, Integer limit) {
+        String errors = null;
+        if (offset == null) {
+            offset = 0;
+        } else if (offset < 0) {
+            errors = "Offset must be greater than or equal to 0.";
+        }
+
+        if (limit == null) {
+            limit = 10;
+        } else if (limit < 1) {
+            if (errors != null) errors += " ";
+            errors += "Limit must be great than or equal to 1.";
+        }
+        if (errors != null) {
+            throw new ValidationException(errors);
+        }
+
```

Comment:
this feels ugly to me. any suggestions?

Reply:
not what you asked for, but typo in the error msg on line 127-- greater than

instead of errors-- what if we just set to default instead of throwing an exception if the numbers are wonky?

Reply:
Wonky numbers mean something went wrong and we'll probably get unexpected results, we should treat it as an exception.